pkg/controller/migplan/migplan_controller.go

```diff
@@ -379,6 +387,48 @@ func (r *ReconcileMigPlan) planSuspended(plan *migapi.MigPlan) error {
 	return nil
 }
 
+func (r *ReconcileMigPlan) AddRunningLabelForPlan(plan *migapi.MigPlan) error {
+	migrations, err := plan.ListMigrations(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	if len(migrations) == 0 {
+		if plan.Labels == nil {
+			plan.Labels = make(map[string]string)
+		}
+		plan.Labels["migplan-running"] = "false"
+		err = r.Update(context.TODO(), plan)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+	}else {
+		for _, m := range migrations {
+			if m.Status.HasCondition(migctl.Running) {
+				if plan.Labels == nil {
```

Comment:
instead of if else condition here both having the update the, I would rather keep a counter `runningMigrations := 0` and then increment it for all running migration, then after the for loop, make a single update call `if runningMigrations > 0`

Reply:
Update the flow.